tests/base.go

```diff
@@ -2,65 +2,56 @@ package tests
 
 import (
 	"bytes"
-	"fmt"
-	"io/ioutil"
-	"log"
+	"crypto/rand"
```

Comment:
please split the base.go file into multiple ones, they can be of the same package, but split them by functionality, as base.go doesn't imply anything